test/services/pacote-service.ts

```diff
@@ -36,6 +38,20 @@ interface ITestCase extends ITestSetup {
 	expectedArgs: any[];
 }
 
+function createPacoteOptions(source: Object): Object {
+		const options: { [index: string]: any } = {};
+		npmconfig.read().forEach((value: any, key: string) => {
```

Comment:
Use a hardcoded config obj here instead of reading the real config from the file system and applying the same business logic in the unit tests.

Reply:
You can't because the actual Pacote driver will be loading the npm config; which it passes back and won't match your hard coded one.  So we have to pass in anything we expect it to get by itself.  :-)

Reply:
Maybe a better way of explaining is it takes the npm configuration, then adds your passed in object to override them.   So if you pass in `{"hi": "blah"}`.  The Pacote driver because it reads the actual npm config will have the other 100 or so settings in addition to the `{"hi": "blah"}` as it loaded it itself.

Reply:
Thanks for the details. As far as I understand, the problem is caused by `libnpmconfig` which is directly used in the Pacote service. In this way, we cannot inject a Stub returning a hardcoded value in the unit tests. If you refactor this part and extract the `npmconfig` related logic into a service, you will be able to mock it in the unit tests and avoid slow disk operations there. Also, have you checked this feature with `yarn`? Are they using the same config file?

Reply:
Yarn will pull the same .npmrc file for server information.  So we should be safe here --  the main values for where servers live and the authentication to talk with them will be pulled from the ~.npmrc file.   

As for moving things into a service and mocking it -- you totally lost me.   Pretend I'm 2yo; and let me know exactly want you need and I'll get it done.   :grinning:    I'll be rebasing this onto Release, and we can move the discussing over their.  :grinning:

Reply:
#4919 is the PR against Release